app/_how-tos/configure-oidc-with-cert-bound-tokens.md

```diff
@@ -0,0 +1,352 @@
+
+cat > keycloak.ext <<EOF
+authorityKeyIdentifier=keyid,issuer
+basicConstraints=CA:FALSE
+keyUsage = digitalSignature, keyEncipherment
+extendedKeyUsage = serverAuth
+subjectAltName = @alt_names
+
+[alt_names]
+DNS.1 = $HOSTNAME
+EOF
+
+openssl x509 -req \
+  -in keycloak.csr \
+  -CA rootCA.crt -CAkey rootCA.key -CAcreateserial \
+  -out keycloak.crt -days 365 -sha256 -extfile keycloak.ext
+```
+{% endcapture %}
+{{ server-cert | indent: 3 }}
+This is used to authenticate with Keycloak and to consume the API with access token. The generated CN must adhere to the pre-defined pattern for Keycloak validation.
+1. Build a PKCS#12 keystore file:
+   ```sh
+   openssl pkcs12 -export \
+     -in keycloak.crt \
+     -inkey keycloak.key \
+     -certfile rootCA.crt \
+     -out keycloak-keystore.p12 \
+     -name keycloak \
+     -passout pass:$PKCS12_PASSWORD
+   ```
+1. Import the root certificates to the `.p12` file:
+{% capture "import-root" %}
+```sh
+keytool -importkeystore \
+  -deststorepass $KEYSTORE_PASS \
+  -destkeypass $KEYSTORE_PASS \
+  -destkeystore keycloak-keystore.jks \
+  -srckeystore keycloak-keystore.p12 \
+  -srcstoretype PKCS12 \
+  -srcstorepass $PKCS12_PASSWORD \
+  -alias keycloak
+```
+{% endcapture %}
+{{ import-root | indent: 3 }}
+
+1. Configure Keycloak to trust certificates signed by the CA:
+{% capture "keycloak-trust" %}
+```sh
+keytool -import -alias rootca \
+  -keystore keycloak-truststore.p12 \
+  -storetype PKCS12 \
+  -file rootCA.crt \
+  -storepass "$PKCS12_PASSWORD"
+
+keytool -list -keystore keycloak-truststore.p12 -storepass "$PKCS12_PASSWORD"
+```
+{% endcapture %}
+{{ keycloak-trust | indent: 3 }}
+Type `y` when prompted to trust the certificate. The Keycloak server presents this certificate to the client.
+
+## Configure Keycloak
+
+1. In a new terminal window, export your trust store password and hostname:
+   ```sh
+   export PKCS12_PASSWORD='YOUR-PASSWORD'
+   export HOSTNAME='YOUR-KEYCLOAK-HOSTNAME'
+   ```
+
+1. Then, start Keycloak in Docker:
+   
+   {:.danger}
+   > **Important:** In this tutorial, use your DNS hostname in place of `$HOSTNAME`.
+{% capture "keycloak-docker" %}
+```sh
+docker run \
+  -p 9443:9443 \
+  --network kong-quickstart-net \
+  -v $(pwd)/oidc/certs:/opt/keycloak/ssl \
+  -e KC_BOOTSTRAP_ADMIN_USERNAME=admin \
+  -e KC_BOOTSTRAP_ADMIN_PASSWORD=admin \
+  --name keycloak \
+  quay.io/keycloak/keycloak \
+  start \
+  --https-port=9443 \
+  --https-certificate-file=/opt/keycloak/ssl/keycloak.crt \
+  --https-certificate-key-file=/opt/keycloak/ssl/keycloak.key \
+  --https-trust-store-file=/opt/keycloak/ssl/keycloak-truststore.p12 \
+  --https-trust-store-password=$PKCS12_PASSWORD \
+  --https-client-auth=request \
+  --hostname=$HOSTNAME
+```
+{% endcapture %}
+{{ keycloak-docker | indent: 3 }}
+
+1. Open the Keycloak admin console.
+   The default URL of the console is `https://$HOSTNAME:8080/admin/master/console/`.
```

Comment:
```suggestion
   The default URL of the console is `https://$HOSTNAME:9443/admin/master/console/`.
```
Shouldn't it be 9443 in this case? You mention that as well in a following step.